engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -0,0 +1,310 @@
+      }
+    }
+  }
+
+  // Represents a scatter collection for a call in the execution store
+  case class CollectorKey(scope: Call) extends JobKey {
+    override val index = None
+    override val attempt = 1
+    override val tag = s"Collector-${scope.unqualifiedName}"
+  }
+
+  case class WorkflowExecutionException(override val throwables: List[Throwable]) extends ThrowableAggregation {
+    override val exceptionContext = s"WorkflowExecutionActor"
+  }
+
+  def props(workflowId: WorkflowId, workflowDescriptor: EngineWorkflowDescriptor): Props = Props(WorkflowExecutionActor(workflowId, workflowDescriptor))
+}
+
+final case class WorkflowExecutionActor(workflowId: WorkflowId, workflowDescriptor: EngineWorkflowDescriptor) extends LoggingFSM[WorkflowExecutionActorState, WorkflowExecutionActorData] {
+
+  import WorkflowExecutionActor._
+  import lenthall.config.ScalaConfig._
+
+  val tag = self.path.name
+  private lazy val DefaultMaxRetriesFallbackValue = 10
+
+  // TODO: We should probably create a trait which loads all the configuration (once per application), and let classes mix it in
+  // to avoid doing ConfigFactory.load() at multiple places
+  val MaxRetries = ConfigFactory.load().getIntOption("system.max-retries") match {
+    case Some(value) => value
+    case None =>
+      log.warning(s"Failed to load the max-retries value from the configuration. Defaulting back to a value of `$DefaultMaxRetriesFallbackValue`.")
+      DefaultMaxRetriesFallbackValue
+  }
+
+  private val factories = TryUtil.sequenceMap(workflowDescriptor.backendAssignments.values.toSeq map { backendName =>
+    backendName -> CromwellBackends.shadowBackendLifecycleFactory(backendName)
+  } toMap) recover {
+    case e => throw new RuntimeException("Could not instantiate backend factories", e)
+  } get
+
+  private val configs = TryUtil.sequenceMap(workflowDescriptor.backendAssignments.values.toSeq map { backendName =>
+    backendName -> BackendConfiguration.backendConfigurationDescriptor(backendName)
+  } toMap) recover {
+    case e => throw new RuntimeException("Could not instantiate backend configurations", e)
+  } get
+
+  // Initialize the StateData with ExecutionStore (all calls as NotStarted) and SymbolStore
+  startWith(
+    WorkflowExecutionPendingState,
+    WorkflowExecutionActorData(
+      workflowDescriptor,
+      executionStore = buildExecutionStore(),
+      outputStore = OutputStore.empty))
+
+  private def buildExecutionStore(): ExecutionStore = {
+    val workflow = workflowDescriptor.backendDescriptor.workflowNamespace.workflow
+    // Only add direct children to the store, the rest is dynamically created when necessary
+    val callExecutions = Scope.collectCalls(workflow.children) map { BackendJobDescriptorKey(_, None, 1) -> NotStarted }
+    val scatterExecutions = Scope.collectScatters(workflow.children) map { ScatterKey(_) -> NotStarted }
```

Comment:
I feel like this algorithm can be a little better (in the category of "thinking out loud").  I'm not a fan of the `collectCalls` and `collectScatters` methods, I think they're mostly a relic from when the `Scope` class was first created.

Maybe something like

```scala
val keys = workflow.children map {
  case call: Call => BackendJobDescriptorKey(call, None, 1)
  case scatter: Scatter => ScatterKey(scatter)
}

ExecutionStore(keys.map(_ -> NotStarted).toMap)
```